freqtrade/freqtradebot.py

```diff
@@ -690,46 +690,77 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
         is enabled.
         """
 
-        result = False
+        logger.debug('Handling stoploss on exchange %s ...', trade)
+
         try:
-            # If trade is open and the buy order is fulfilled but there is no stoploss,
-            # then we add a stoploss on exchange
-            if not trade.open_order_id and not trade.stoploss_order_id:
-                if self.edge:
-                    stoploss = self.edge.stoploss(pair=trade.pair)
-                else:
-                    stoploss = self.strategy.stoploss
+            # First we check if there is already a stoploss on exchange
+            stoploss_order = self.exchange.get_order(trade.stoploss_order_id, trade.pair) \
+                if trade.stoploss_order_id else None
+        except DependencyException as exception:
+            logger.warning('Unable to fetch stoploss order: %s', exception)
+
+        # If trade open order id does not exist,
+        # it means buy order is fulfilled
+        buy_order_fulfilled = not trade.open_order_id
 
-                stop_price = trade.open_rate * (1 + stoploss)
+        # Limit price threshold
+        # This is the limit price percentage below which you don't want to sell
+        # 0.99 is arbitrary here.
+        # As limit price should always be below price
+        limit_price_pct = 0.99
+
+        # If buy order is fulfilled but there is no stoploss,
+        # then we add a stoploss on exchange
+        if (buy_order_fulfilled and not stoploss_order):
```

Comment:
This will fail with `NameError - stoploss_order is not defined` if get_order raises a `DependencyException`.
You need to define stoploss_order before the try block as none to prevent that from happening.

The fact that this bug is here - but all tests pass tells me that there is a test missing ...

Reply:
awesome !